fix: cli parsing for compression and encryption parameters

fabriceclementz opened the pull request:
This PR closes #53

Compression and encryption was always override by the `destination` section in the config file. https://github.com/Qovery/replibyte/compare/main...fabriceclementz:fix-encryption?expand=1#diff-70dba71af950fe9043b343b216d6e7d1a9e8b1c85b06bbc8b060aad5111e887fL86

For the restore commands, it makes sense to check if the backup is compressed and encrypted directly from the backup manifest and not from the config file. This PR removes the encryption_key and compression option in the destination config.

The encryption key is always read from the source configuration. As the encryption key needs to be the same when encrypting or decrypting, we could add the encryption_key at the top level of the yaml file too ? WDYT @evoxmusic

evoxmusic commented:
Thank you Fab. I think it's a good idea to put the encryption key at the top level since it will be the same for source and destination. 👌

evoxmusic commented:
I think the only thing missing is updating the README.md

fabriceclementz commented:
I've updated the README and moved the encryption_key at the top level.

A little bug was introduced in the telemetry commit by [returning `Ok(())` for every commands](https://github.com/Qovery/replibyte/commit/e36695a2283332d8523ce693c55a15ecbbf9e245#diff-70dba71af950fe9043b343b216d6e7d1a9e8b1c85b06bbc8b060aad5111e887fR137) the consequence was the error message wasn't printed anymore. To fix it, I've extracted some logic in a run function. 

As always, thx for your feedback @evoxmusic 🚀
